### PR TITLE
Small fixes for the periodogram page

### PR DESCRIPTION
**Pull request by profjsb**
- Handle the case where there is not much data observed over a very long time period, producing a more robust delta_t
- Handle the case when there is only a few data points (fixes what @dmitryduev found with a single photometry point). In that case, do not render the periodogram.
- ensure that we're only doing as many as 300,000 calculations (~<5 seconds in a modern browser)
- render the new light curve after resetting parameters.

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
Looks good, thanks @profjsb!

File: static/js/components/Periodogram.jsx [hunk 1]
```diff
@@ -141,11 +154,26 @@ function GLS(t_data_uf, y_data_uf, kwa) {
...
+    quantileSeq(diffs, 0.75, false),
+    0.05 * ofac
+  );
+  let fbeg = parseFloat(kwargs.fbeg || (df * ofac) / 2);
+  let fend = parseFloat(kwargs.fend || (0.5 * ofac) / delta_t);
+  let nf = Math.floor((fend - fbeg) / df) + 1; // size of frequency grid
+
+  // ensure that we're only doing as many as 300,000 calculations (~<5 seconds)
+  // bump down the frequency binning (df) and shrink the start and ending limits
+  while (nf > 3e5) {
```
Comment (profjsb) [inline static/js/components/Periodogram.jsx:171]: Until we have a client-side graphics card implementation, limit the total number of frequencies to 300,000. This takes at most ~5 seconds on a modern system/browser.

**Comment by dmitryduev** [comment]
Waiting on the CI, will merge once green.

**Comment by dmitryduev** [comment]
The only failure on the CI is in `test_periodogram` -- could you please look into it @profjsb?

File: skyportal/tests/frontend/test_periodogram.py [hunk 2]
```diff
@@ -9,40 +12,66 @@
...
+    driver.get(f"/source/{obj_id}/periodogram")
 
     # If the bestp id shows up then a period was found
     best_period = "//span[contains(@data-testid, 'bestp')]"
-    driver.wait_for_xpath(best_period)
+    bestp = driver.wait_for_xpath(best_period)
+
+    # did we find the right period?
+    bestp_flt = float(bestp.text)
+    npt.assert_almost_equal(bestp_flt, P, decimal=1)
```
Comment (dmitryduev) [inline skyportal/tests/frontend/test_periodogram.py:74]: nice!

**Review by dmitryduev — APPROVED** [review]

**Pull request merged**